### PR TITLE
Make oak_abi a standalone crate

### LINKED ISSUE #971: Rust cache invalidated between native and Wasm builds
To reproduce:

- `cargo build --release --target=wasm32-unknown-unknown --package=aggregator`
  + `Finished release [optimized] target(s) in 43.39s` (slow)
- `cargo build --release --target=wasm32-unknown-unknown --package=aggregator`
  + `Finished release [optimized] target(s) in 0.14s` (fast)
- `cargo build --release --package=aggregator_backend`
  + `Finished release [optimized] target(s) in 1m 15s` (slow)
- `cargo build --release --package=aggregator_backend`
  + `Finished release [optimized] target(s) in 0.15s` (fast)
- `cargo build --release --target=wasm32-unknown-unknown --package=aggregator`
  + `Finished release [optimized] target(s) in 43.40s` (expected: fast; actual: slow again)

I am guessing that switching compilation target causes some dependencies to be rebuilt (looks like `libc` may be a potential culprit).

@rbehjati could you look into it? It will become more and more relevant as we switch to the Rust version of `main` for the `oak_loader`, as we will interleave compiling native and Wasm code even more often then.

### PR DESCRIPTION
**Pull request by rbehjati**
Ref #971 

# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by tiziano88** [comment]
will take a look shortly, but yes, crates live in a global namespace, so `abi` is not specific enough (as a matter of fact, even `oak_abi` was already taken, so we probably need something even more specific)

**Comment by daviddrysdale** [comment]
Aside: [`oak_abi`](https://crates.io/crates/oak_abi) looks free at the moment, but [`oak_runtime`](https://crates.io/crates/oak_runtime) isn't.

**Comment by rbehjati** [comment]
I changed the name to `oak_abi` for now.

File: Cargo.toml [hunk 1]
```diff
@@ -37,7 +36,6 @@ members = [
 [patch.crates-io]
 # Oak.
 oak = { path = "sdk/rust/oak" }
-oak_abi = { path = "oak/server/rust/oak_abi" }
```
Comment (tiziano88) [inline Cargo.toml:40]: Why not keep this and just redirect it to the new location?

Comment (rbehjati) [reply Cargo.toml:40]: Oh! This works! Thanks.

**Pull request merged**